File: 5-longest-palindromic-substring/longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        result = ""
        resultLen = 0
        for i in range(len(s)):
            l = i
            r = i
            while 0 <= l and r < len(s) and s[l] == s[r]:
                if ((r+1)-l) >= resultLen:
                    result = s[l:r+1]
                    resultLen = (r+1)-l
                l -= 1
                r += 1
            l = i
            r = i + 1
            while 0 <= l and r < len(s) and s[l] == s[r]:
                if ((r+1)-l) >= resultLen:
                    print(s[l:r+1])
                    result = s[l:r+1]
                    resultLen = (r+1)-l
                l -= 1
                r += 1
        return result

File: 5-longest-palindromic-substring/longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        string result = "";
        int resultLen = 0;
        for (int i  = 0; i < s.length(); i++){
            int l = i, r = i;
            while (0<=l && r < s.length() && s[l] == s[r]){
                if((r + 1) - l >= resultLen){
                    resultLen = (r + 1) - l;
                    result = s.substr(l, (r + 1) - l); 
                }
                l -= 1;
                r += 1;
            }
            l = i, r = i + 1;
            while (0<=l && r < s.length() && s[l] == s[r]){
                if((r + 1) - l >= resultLen){
                    resultLen = (r + 1) - l;
                    result = s.substr(l, (r + 1) - l); 
                }
                l -= 1;
                r += 1;
            }
        }
        return result;
    }
};